### DISCUSSION TITLE
How to start skywalking with h2 storage in Helm Chart

### DISCUSSION THREAD

**Discussion by andrewgkew** [apache/skywalking#7501] [Q&A] [answered]
Please answer these questions before submitting your issue.

- Why do you submit this issue?
- [x] Question or discussion
- [ ] Bug
- [ ] Requirement
- [ ] Feature or performance improvement

___
### Question
- What do you want to know?
How do I start skwalking up using h2 as the backend
___
### Bug
- Which version of SkyWalking, OS, and JRE?
Skywalking version: 8.7.0
OS: MacOs 10.13.6
JRE: 8

- Which company or project?
QuadCorps Ltd

- What happened?
If possible, provide a way to reproduce the error. e.g. demo application, component version.

I am trying to run skywalking on a local cluster (Docker Engine) using your helm chart to install it
This is the command that I ran

helm install "${SKYWALKING_RELEASE_NAME}" skywalking -n "${SKYWALKING_RELEASE_NAMESPACE}" \
  --set oap.image.tag=8.7.0-es7 \
  --set oap.storageType=h2\
  --set ui.image.tag=8.7.0 \
  --set elasticsearch.enabled=false

___
### Requirement or improvement
- Please describe your requirements or improvement suggestions.

The UI pod comes up but the OAP does not, it moans about tables not being present.
By default it looks like OAP is started in no-init mode set by its env variables for the JAVA_OPTS
Looking at the OAP pod logs its moaning about tables not existing, which makes me think h2 needs to be initialized

How do I do that? Override the JAVA_OPTS with -Dmode=init or -Dmode=init-only
I cant find anything in your docs that describes how to get this to work using h2

Also is the OAP image tag *-es7 correct even when using h2 and not elastic search?

**Comment by kezhenxu94** [comment, marked as answer]
Here is an example to install SkyWalking in Helm Chart with storage `h2`:

```
helm -n istio-system install skywalking skywalking --set oap.storageType='h2' --set ui.image.tag=8.5.0 --set oap.image.tag=8.5.0-es7 --set oap.replicas=1 --set oap.env.JAVA_OPTS='-Dmode=' --set elasticsearch.enabled=false
```

The key point is `-Dmode=` to unset the `mode` property.

> I cant find anything in your docs that describes how to get this to work using h2

H2 storage is only for demonstration, and Helm Chart is usually for production usage, that's why we don't have many docs about this.

> Also is the OAP image tag *-es7 correct even when using h2 and not elastic search?
 
Yes

**Reply by wu-sheng** [reply]
One more, if you want to use H2 as same mode as ElasticSearch, you need a separate pod of H2 running in server mode. Then set jdbc url to the pod address.

**Reply by andrewgkew** [reply]
Thanks @kezhenxu94 I will give that a try
And thanks @wu-sheng that makes sense 
Let me give both a try

**Reply by andrewgkew** [reply]
Do I set the h2 url with this parameter

`--set storage.h2.url=jdbc:h2:mem:<pod-name>`

**Reply by wu-sheng** [reply]
Take a reference from https://skywalking.apache.org/docs/main/v8.7.0/en/setup/backend/configuration-vocabulary/#configuration-vocabulary please. I think system env variable is `SW_STORAGE_H2_URL`

**Reply by wu-sheng** [reply]
> --set storage.h2.url=jdbc:h2:mem:<pod-name>

`mem` is not correct. http://www.h2database.com/html/features.html#database_url

**Reply by andrewgkew** [reply]
Thanks @wu-sheng - That vocab table is exactly what I have been looking for. I didnt realize the table scrolled to the right with the system vars, that is perfect.

And yes thanks for the h2 docs - I can see the mem is for in memory only I will give this a go

I tried the option with h2 running on OAP and that worked a treat so thank you for your help 👍
